versions/models.py

```diff
@@ -88,17 +100,15 @@ def next_version(self, object):
         if object.version_end_date == None:
             return object
         else:
-            try:
-                next = self.get(
-                    Q(identity=object.identity),
-                    Q(version_start_date=object.version_end_date))
-            except MultipleObjectsReturned as e:
-                raise MultipleObjectsReturned(
-                    "next_version couldn't uniquely identify the next version of object " + str(
-                        object.identity) + " to be returned\n" + str(e))
-            except ObjectDoesNotExist as e:
+            next = self.filter(
+                Q(identity=object.identity),
+                Q(version_start_date__gte=object.version_end_date)
+            ).order_by('version_start_date').first()
```

Comment:
Just to be sure, you meant to implement it this way: This means, that 2 versions of a same object can exist at the same time, and CleanerVersion wouldn't complain.